core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -561,9 +561,6 @@ public void refresh(String source) throws EngineException {
         try (ReleasableLock lock = readLock.acquire()) {
             ensureOpen();
             searcherManager.maybeRefreshBlocking();
-        } catch (AlreadyClosedException e) {
```

Comment:
Also, note that this refresh call can come from, for example, `InternalEngine#writeIndexingBuffer` from `IndexShard#writeIndexingBuffer` which has a general catch block which will swallow the `AlreadyClosedException`!